src/Xtools/Repository.php

```diff
@@ -134,6 +138,29 @@ public function isLabs()
     }
 
     /**
+     * Make a request to the XTools API. This requires that app.multithread.api_url be set.
+     * @param string $endpoint Relative path to endpoint with relevant query parameters.
+     * @param bool $async Set to true to asynchronously query and return a promise.
+     * @return mixed|GuzzleHttp\Promise\Promise
+     */
+    public function queryXTools($endpoint, $async = false)
+    {
+        if (!$this->apiConnection) {
+            $this->apiConnection = $this->container->get('guzzle.client.xtools');
+        }
+
+        // Ensure there's a trailing slash.
+        $apiRoot = trim($this->container->getParameter('app.multithread.api_url'), '/');
```

Comment:
So am I thinking correctly: I think this should default to the XTools base URL, if `app.multithread.api_url` isn't set (or is set to the default), so that it can still hit the API.